video/out/vulkan/context.c

```diff
@@ -263,6 +263,15 @@ bool ra_vk_ctx_init(struct ra_ctx *ctx, struct mpvk_ctx *vk)
         .clipped = true,
     };
 
+    /* Because in Wayland clients render whenever they receive a callback from the
+     * compositor, and the fact that the compositor usually stops sending callbacks
+     * once the surface is no longer visible, using FIFO here would mean the entire
+     * player would block on acquiring swapchain images. Hence, use MAILBOX to
+     * guarantee that there'll always be a swapchain image and the player won't
+     * block waiting on those */
+    if (ctx->vo->wayland)
+        p->protoInfo.presentMode = VK_PRESENT_MODE_MAILBOX_KHR;
```

Comment:
Having such special cases should be avoided at all cost. It should be done in the wayland specific backend instead.